### PR TITLE
connecting React and Redux

### PR DESCRIPTION
**Pull request by Abdulwhbhussain**
Project requirements
Wrap your <App> with the <Provider> component from react-redux
Pass your configured store into the <Provider>
Create an initialState variable for your booksSlice, which will include the following:
// Initial state:
[
  {
    "item_id": "item1",
    "title": "The Great Gatsby",
    "author": "John Smith",
    "category": "Fiction"
  },
  {
    "item_id": "item2",
    "title": "Anna Karenina",
    "author": "Leo Tolstoy",
    "category": "Fiction"
  },
  {
    "item_id": "item3",
    "title": "The Selfish Gene",
    "author": "Richard Dawkins",
    "category": "Nonfiction"
  }
]
Display your books, received from the slice, in a reusable component
Dispatch actions using useDispatch
Add a <Button> component, which includes:
An event handler that adds a book to the books array (with attributes id, title and author)
Add a <Button> component, which includes:
An event handler that removes a book from the books array (by id)

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const initialState = {
   categories: [],
```
Comment (bobb-Rob) [inline src/redux/categories/categoriesSlice.js:4]: - [ ] Based on the project requirement, you are suppose to add the initial book object to the BookSlice as shown below 👇🏾 


```javascript
```javascript
import { createSlice } from "@reduxjs/toolkit"

const initialState = {
  books: [
    {
      item_id: "item1",
      title: "The Great Gatsby",
      author: "John Smith",
      category: "Fiction",
    },
    {
      item_id: "item2",
      title: "Anna Karenina",
      author: "Leo Tolstoy",
      category: "Fiction",
    },
    {
      item_id: "item3",
      title: "The Selfish Gene",
      author: "Richard Dawkins",
      category: "Nonfiction",
    },
  ],
}

export const booksSlice = createSlice({
  name: "books",
  initialState,
  reducers: {
    addBook: (state, action) => {
      state.books = [...state.books, action.payload]
    },
    removeBook: (state, action) => {
      state.books = state.books.filter((book) => book.id !== action.id)
    },
  },
})

export const { addBook, removeBook } = booksSlice.actions

export default booksSlice.reducer
```
```

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @Abdulwhbhussain, 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (peterrobert) [reply src/redux/categories/categoriesSlice.js:4]: This change has been done well

**Review by peterrobert — APPROVED** [review]
Hi @Abdulwhbhussain, 


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by topeogunleye** [comment]
Hi @Abdulwhbhussain, 

Overall, the student has done a great job implementing the required functionality using React and Redux. The components responsible for adding and removing books are functioning as intended. It's good to see the correct usage of React state, Redux state management, and event handling.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

dwayne-the-rock

## (Highlights) Good Points: 💪 
Code Review Summary:

🔍 Overall, the student's project fulfills the specified requirements. Here are the highlights and feedback:

✅ The main React component is wrapped with the `<Provider>` component from react-redux, as required.

✅ The Redux store has been passed into the `<Provider>`, meeting the requirement.

✅ An initialState has been added to the booksSlice, satisfying the requirement.

✅ The books are accessed from inside the `<BookList>` component with `useSelector`, ensuring the correct usage of Redux state.

Adding a book:
✅ The `<NewBookForm>` component has local state for title and author, updated through the onChange event.

✅ The `<NewBookForm>` component includes a `<Button>` with an event handler to add a book to the books array, providing the title and author attributes.

✅ The `<NewBookForm>` component is correctly exported with `export default NewBookForm`.

Removing a book:
✅ The component responsible for rendering each book (likely `<Book>`) has a `<Button>` with an event handler to remove a book from the books array, using the id attribute.

Final check:
✅ The student's app allows adding and removing books successfully, as expected.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

You can also consider:
👉 One optional suggestion for improvement is to follow the convention for naming your branch. Usually, branch names should use the past tense verb to indicate that the feature has been implemented. For example, instead of `connecting-react-redux`, you could use `connected-react-redux`. This would make your branch name more consistent and clear for future pull requests.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**